nds-org/kubeadm-terraform
# Weave support

# Problem
Fixes #3 - There is no current configuration option for selecting the desired type of Pod Network.

# Approach
This PR adds an additional value to `variables.tf` for the `pod_network_type`. Currently supported options are `flannel` or `weave`.

NOTE: Flannel will be deployed by default.

# How to test

NOTE: This PR requires the changes from https://github.com/cheese-hub/kubeadm-bootstrap/pull/1 in order to properly function

0. Specify `weave` as the `pod_network_type` in your `variables.tf`
1. Execute `terraform init && terraform apply`
2. Wait for your cluster to come up
3. SSH into the master node
4. Execute `kubectl get pods -n kube-system`
    * You should see that weave is now running, instead of flannel
